Receptai.client/components/Banner/Banner.vue

```diff
@@ -2,46 +2,140 @@
 import axios from "axios";
 import RecipeBannerComponent from "@/components/Banner/RecipeBannerComponent.vue";
 
-const config = useRuntimeConfig();
-
 interface Recipe {
     id: number;
     name: string;
     shortDescription: string;
+    author: Author;
+    dateCreated: string;
+    dateModified: string | null;
     previewImage: string;
+    tutorialVideo?: string;
+    ingredients: Ingredients[];
+    instructions: string[];
+    tags: string[];
+    categories: Category[];
+    minutesToPrepare: number;
+    portions: number;
+}
+
+interface Author {
+    name: string;
+}
+
+interface Ingredients {
+    purpose: string;
+    ingredients: Ingredient[];
+}
+
+interface Ingredient {
+    name: string;
+    quantity: number;
+    unit: string;
+}
+
+interface Category {
+    id: number;
+    name: string;
+    description: string | null;
+    previewImageUrl: string | null;
+}
+
+const recipeList = ref<Recipe[]>([]);
+const bannerOffset = ref(0);
+const bannerComponentWidth = 192;
+const bannerWidth = ref(0);
+let dragStartX = 0;
+let isDragging = false;
+let moveBannersInterval: NodeJS.Timeout | null = null;
+
+async function fetchRecipes() {
+    try {
+        const response = await axios.get("/api/v1/recipe/list?page=0&sortBy=dateCreated&sortAsc=false");
+        recipeList.value = response.data.elements.slice(0, 20);
+    } catch (error) {
+        console.error("Error fetching recipes", error);
+    }
+}
+
+fetchRecipes();
+
+function startDrag(event: MouseEvent | TouchEvent): void {
+    isDragging = true;
+    dragStartX = getEventX(event);
+
+    if (moveBannersInterval) {
+        clearInterval(moveBannersInterval);
+        moveBannersInterval = null;
+    }
+}
+
+function drag(event: MouseEvent | TouchEvent): void {
+    if (!isDragging) return;
+    const currentX = getEventX(event);
+    const delta = currentX - dragStartX;
+    bannerOffset.value += delta;
+    dragStartX = currentX;
+}
+
+function endDrag(): void {
+    isDragging = false;
+    if (!moveBannersInterval) {
+        moveBannersInterval = setInterval(nextSlide, 2000);
+    }
```

Comment:
I don't know if this is the intendet approach, but moveBannersInterval starts to move recipes only when banner carusel has been draged. I would imagine, that timer should start imediatly and stop when dragging starts.